src/avif.c

```diff
@@ -721,6 +721,54 @@ avifBool avifCleanApertureBoxConvertCropRect(avifCleanApertureBox * clap,
     return AVIF_TRUE;
 }
 
+// ---------------------------------------------------------------------------
+
+avifBool avifAreGridDimensionsValid(avifPixelFormat yuvFormat, uint32_t columns, uint32_t rows, uint32_t tileW, uint32_t tileH, avifDiagnostics * diag)
+{
+    // The rules below only apply to grid images. Single-cell images do not have these constraints.
+    if ((columns == 1) && (rows == 1)) {
+        return AVIF_TRUE;
+    }
+
+    // ISO/IEC 23000-22:2019, Section 7.3.11.4.2:
+    //   - the tile_width shall be greater than or equal to 64, and should be a multiple of 64
+    //   - the tile_height shall be greater than or equal to 64, and should be a multiple of 64
+    // The "should" part is ignored here.
+    if ((tileW < 64) || (tileH < 64)) {
+        avifDiagnosticsPrintf(diag,
+                              "[Strict] Grid image tile width (%u) or height (%u) cannot be smaller "
```

Comment:
Is this the case for the "[Strict]" in [avifCropRectIsValid()](https://github.com/AOMediaCodec/libavif/blob/d9cffc5f46b62aeff46eebf51449726386d6c485/src/avif.c#L531)? I do not see how this specific check can be turned off, except not using `avifenc --crop`.

Reply:
I looked into this. I found that the policy on "[Strict]" cannot be followed perfectly because right now only `avifDecoder` has a `strictFlags` field. So I will only show how the policy on "[Strict]" is followed by `avifCropRectIsValid()` on the decoder side.

`avifCropRectIsValid()` is called by `avifCropRectConvertCleanApertureBox()`. Note that all the diagnostic messages in `avifCropRectIsValid()` and `avifCropRectConvertCleanApertureBox()` have the "[Strict]" prefix.

The decoder calls `avifCropRectConvertCleanApertureBox()` as follows:

```
    if (strictFlags & AVIF_STRICT_CLAP_VALID) {
        const avifProperty * clapProp = avifPropertyArrayFind(&item->properties, "clap");
        if (clapProp) {
            const avifProperty * ispeProp = avifPropertyArrayFind(&item->properties, "ispe");
            if (!ispeProp) {
                avifDiagnosticsPrintf(diag,
                                      "[Strict] Item ID %u is missing an ispe property, so its clap property cannot be validated",
                                      item->id);
                return AVIF_RESULT_BMFF_PARSE_FAILED;
            }

            avifCropRect cropRect;
            const uint32_t imageW = ispeProp->u.ispe.width;
            const uint32_t imageH = ispeProp->u.ispe.height;
            const avifPixelFormat av1CFormat = avifCodecConfigurationBoxGetFormat(&av1CProp->u.av1C);
            avifBool validClap = avifCropRectConvertCleanApertureBox(&cropRect, &clapProp->u.clap, imageW, imageH, av1CFormat, diag);
            if (!validClap) {
                return AVIF_RESULT_BMFF_PARSE_FAILED;
            }
        }
    }
```

Note that it is inside an if statement that checks `strictFlags & AVIF_STRICT_CLAP_VALID`.